server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -722,6 +724,55 @@ public String toString() {
         };
     }
 
+    /**
+     * Updates settings that depend on each other.
+     */
+    static <A, B, C> AbstractScopedSettings.SettingUpdater<Triplet<A, B, C>> compoundUpdater(
+        final TriConsumer<A, B, C> consumer,
+        final TriConsumer<A, B, C> validator,
+        final Setting<A> aSetting,
+        final Setting<B> bSetting,
+        final Setting<C> cSetting,
+        Logger logger
+    ) {
+        final AbstractScopedSettings.SettingUpdater<A> aSettingUpdater = aSetting.newUpdater(null, logger);
+        final AbstractScopedSettings.SettingUpdater<B> bSettingUpdater = bSetting.newUpdater(null, logger);
+        final AbstractScopedSettings.SettingUpdater<C> cSettingUpdater = cSetting.newUpdater(null, logger);
+        return new AbstractScopedSettings.SettingUpdater<Triplet<A, B, C>>() {
+            @Override
+            public boolean hasChanged(Settings current, Settings previous) {
+                return aSettingUpdater.hasChanged(current, previous)
+                    || bSettingUpdater.hasChanged(current, previous)
+                    || cSettingUpdater.hasChanged(current, previous);
+            }
+
+            @Override
+            public Triplet<A, B, C> getValue(Settings current, Settings previous) {
+                A valueA = aSettingUpdater.getValue(current, previous);
+                B valueB = bSettingUpdater.getValue(current, previous);
+                C valueC = cSettingUpdater.getValue(current, previous);
+                validator.accept(valueA, valueB, valueC);
+                return new Triplet<>(valueA, valueB, valueC);
+            }
+
+            @Override
+            public void apply(Triplet<A, B, C> value, Settings current, Settings previous) {
+                if (aSettingUpdater.hasChanged(current, previous)) {
+                    logSettingUpdate(aSetting, current, previous, logger);
+                }
+                if (bSettingUpdater.hasChanged(current, previous)) {
+                    logSettingUpdate(bSetting, current, previous, logger);
+                }
```

Comment:
should we add a block here for cSettingUpdater?

Reply:
Yes, added the block for cSetting logUpdate as well. Thanks for catching this!